auto_dev/data/repo/templates/autonomy/install.sh

```diff
@@ -38,82 +35,97 @@ get_download_url() {
             protolint_arch="arm64"
             ;;
         *)
-            echo "Unsupported architecture: $arch" >&2
+            echo "Unsupported architecture: ${arch}" >&2
             return 1
             ;;
     esac
 
-    if [ "$tool" = "protoc" ]; then
-        if [ "$os" = "Darwin" ]; then
+    if [ "${tool}" = "protoc" ]; then
+        if [ "${os}" = "Darwin" ]; then
             echo "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-osx-universal_binary.zip"
         else
             echo "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-${protoc_os}-${protoc_arch}.zip"
         fi
-    elif [ "$tool" = "protolint" ]; then
+    elif [ "${tool}" = "protolint" ]; then
         echo "https://github.com/yoheimuta/protolint/releases/download/v${PROTOLINT_VERSION}/protolint_${PROTOLINT_VERSION}_${protolint_os}_${protolint_arch}.tar.gz"
     else
-        echo "Unknown tool: $tool" >&2
+        echo "Unknown tool: ${tool}" >&2
         return 1
     fi
 }
 
 # Function to download and install a tool
 install_tool() {
-    local tool=$1
+    local tool="$1"
     local url
-    local venv_dir=$(echo $(poetry run which python) | sed 's|\(.*\)/python|\1|')
-    url=$(get_download_url "$tool") || return 1
+    local venv_dir
+
+    # Get virtualenv path from poetry env info
+    printf "Getting virtualenv path from poetry... "
```

Comment:
this check should be way earlier imo? shouldnt even be downloading if the poetry setup fails?

Reply:
this is simply a replacement for `    local venv_dir=$(echo $(poetry run which python) | sed 's|\(.*\)/python|\1|')` which errors due to the use of `which` (not sure why)

Reply:
found a simpler replacement in 346c18cbf534d5d22d58dbc08636514488608024